### ISSUE TITLE
JMP (IND) is 5 cycles only - not 6 - on 6502

### ISSUE THREAD

**Issue by Archange427** [AppleWin/AppleWin#264]
I hope I don't make a mistake but :
according to documentations (and my observations with a "true" Apple IIe 6502), Indirect Jump (opcode $6C) is only 5 cycles - not 6 - on a 6502 CPU.

docs : 
http://archive.6502.org/datasheets/mos_6500_mpu_preliminary_may_1976.pdf 
http://archive.6502.org/datasheets/rockwell_r650x_r651x.pdf

From Eyes/Lichty book ("Programming the 65816..."), it seems that JMP (IND) is 6 cycles only on 65C02
doc : http://archive.6502.org/datasheets/rockwell_r65c00_microprocessors.pdf

**Comment by sicklittlemonkey** [comment]
Confirmed in Sather's Understanding the Apple //e.
(Compare table 4.1 to 4.3, rows 29 & 30.)

Cheers,
Nick.
​

**Comment by tomcw** [comment]
With a Mockingboard in slot 4 (ie. a 6522 mapped at $C400), I tried the following simple program:
300: AD 04 C4 6C 06 03 08 03 AE 04 C4 00

````
.ORG $300
 LDA $C404
 JMP ($306)
; 306:
.BYTE 08 03
 LDX $C404
 BRK
````

Currently AppleWin 1.25.0.3 gives:
* Original //e (6502): A-X delta = 10 cycles
* Enhanced //e (65C02): A-X delta = 10 cycles

On a real original Apple //e (with NMOS 6502) I get these results:
A=14 X=0B
A=E0 X=D7
A=DA X=D1
ie. a consistent (A-X) delta of 9 cycles.

Since LDA ABS is 4 cycles then **JMP (IND) is 5 cycles** for the real NMOS 6502.

**Comment by tomcw** [comment]
Fixed in 614dfc1257cc7fc32fb344be27944f145cdd8935. Closing.
